### ISSUE TITLE
Bad behavior when downloading videos through native YouTube app share function

### ISSUE THREAD

**Issue by exegg** [TeamNewPipe/NewPipe#2838]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
It does
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.
It isn't a crash.

I will post a video since I think it is self explanatory, but I'll summary the few issues.

https://streamable.com/4jyw7

- After selecting a video for download, NewPipe stays opened, frozen, with the youtube screen you had opened (00:23). You have to manually switch from NewPipe to YouTube again.

- After downloading a video, when choosing another one, it brings again the latest video you already downloaded instead the one you are fetching. (00:37). You have to repeat the steps in order to get the actual video you want to download.
Ironically when you cancel, the floating window disappears correctly (so NewPipe doesn't stay open, frozen) and you can continue to browse YouTube.

This behavior wasn't present in past versions. I've been using this feature for as long as I can remember using the app.
I'm using the latest stable version from Github.
Samsung Galaxy J7 2018 (j737a) with android 8.0

**Comment by cool-student** [comment]
This could be related to https://github.com/TeamNewPipe/NewPipe/issues/2773 it sounds like you experience problems with the downloading process.

**Comment by exegg** [comment]
Well yes, it is exactly the same issue described by gdevenyi on that issue. The issue is not present in any previous versions as far as I know it.

**Comment by Nico-late** [comment]
I found a way to solve this issue together with issue #2773  and issue #2797 . Youtube seems to freeze as the NewPipe activity is not correctly closed. I added an if statement in DownloadDialog to close the activity when it's not coming from the app itself: 
```java 
toolbar.setOnMenuItemClickListener(item -> {
            if (item.getItemId() == R.id.okay) {
                prepareSelectedDownload();
                if (getActivity() instanceof RouterActivity){
                    getActivity().finish();
                }
                return true;
            }
            return false;
        });
```
 As it finishes the activity, when downloading a second video, the name is reset so it also fixes issue #2797 .
